exercises/two-bucket/Example.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+
+public enum Bucket
+{
+    One,
+    Two
+}
+
+public class TwoBucketResult
+{
+    public int Moves { get; set; }
+    public Bucket GoalBucket { get; set; }
+    public int OtherBucketContents { get; set; }
+}
+
+public class BucketContainer
+{
+    public BucketContainer(int contents, int capacity)
+    {
+        Contents = contents;
+        Capacity = capacity;
+    }
+
+    public int Contents { get; set; }
+
+    public int Capacity { get; }
+
+    public bool IsEmpty => Contents == 0;
+
+    public bool IsFull => Contents == Capacity;
+
+    public void Fill()
+    {
+        Contents = Capacity;
+    }
+
+    public void Empty()
+    {
+        Contents = 0;
+    }
+
+    public void PourTo(BucketContainer other)
+    {
+        var amount = Math.Min(other.Capacity - other.Contents, Contents);
+        Contents -= amount;
+        other.Contents += amount;
+    }
+
+    public bool CanPourTo(BucketContainer other) => IsFull && !other.IsFull || !IsFull && other.IsEmpty;
```

Comment:
Is there a reason this logic isn't `!IsEmpty && !other.IsFull`? You should be able to pour whenever the current bucket isn't empty and the target bucket isn't full right?